### PR TITLE
Fix dataset formats (cyrilic and spaces in filename)

### PR DESCRIPTION
**Pull request by yasakova-anastasia**
### Summary

Resolves [openvinotoolkit/cvat#2840](https://github.com/openvinotoolkit/cvat/issues/2840)
Resolves [openvinotoolkit/cvat#2922](https://github.com/openvinotoolkit/cvat/issues/2922)
- Added the ability to work with file names containing Cyrillic and spaces

### Checklist

- [x] I submit my changes into the `develop` branch
- [x] I have added description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md)
- [ ] I have updated the [documentation](
  https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly
- [x] I have added tests to cover my changes
- [ ] I have [linked related issues](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/opencv/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below)

```python
# Copyright (C) 2020 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

**Comment by zhiltsov-max** [comment]
Are any other formats affected? Should we add such tests in all formats?

File: datumaro/plugins/voc_format/converter.py [hunk 1]
```diff
@@ -367,8 +367,10 @@ def save_action_lists(self, subset_name, action_list):
         items = {k: True for k in action_list}
         if self._patch and osp.isfile(ann_file):
             self._get_filtered_lines(ann_file, self._patch, subset_name, items)
-        with open(ann_file, 'w') as f:
+        with open(ann_file, 'w', encoding='utf-8') as f:
             for item in items:
+                if 1 < len(item.split()):
```
Comment (zhiltsov-max) [inline datumaro/plugins/voc_format/converter.py:372]: I don't think this solution is right. All the VOC subformats should handle such names without changes. The only problematic subformat should be `Layout`, where only parsing should be changed.

**Review by zhiltsov-max — CHANGES_REQUESTED** [review]

File: datumaro/plugins/imagenet_txt_format.py [hunk 2]
```diff
@@ -49,9 +49,14 @@ def _load_items(self, path):
         items = {}
         with open(path, encoding='utf-8') as f:
             for line in f:
-                item = line.split()
-                item_id = item[0]
-                label_ids = [int(id) for id in item[1:]]
+                item = line.split('\"')
```
[the remark this answers is not in the document]
Comment (yasakova-anastasia) [reply datumaro/plugins/imagenet_txt_format.py:52]: Could you explain how best to do it? Now we go through `if` when we have two characters `"`. In all other cases we go through `else`.

[the remark this answers is not in the document]
Comment (zhiltsov-max) [reply datumaro/plugins/imagenet_txt_format.py:52]: For instance, we could raise an error if there is any unexpected number of quotes - 1 or 3+.

File: datumaro/plugins/voc_format/extractor.py [hunk 3]
```diff
@@ -78,11 +84,13 @@ def _load_annotations(self):
         anno_files = [s for s in dir_items(task_dir, '.txt')
             if s.endswith('_' + osp.basename(self._path))]
         for ann_filename in anno_files:
-            with open(osp.join(task_dir, ann_filename)) as f:
+            with open(osp.join(task_dir, ann_filename), encoding='utf-8') as f:
                 label = ann_filename[:ann_filename.rfind('_')]
                 label_id = self._get_label_id(label)
                 for line in f:
-                    item, present = line.split()
+                    objects = line.split()
```
Comment (zhiltsov-max) [inline datumaro/plugins/voc_format/extractor.py:91]: Consider using `rsplit(maxsplit=1)`

File: datumaro/plugins/camvid_format.py [hunk 4]
```diff
@@ -61,6 +61,7 @@ class CamvidPath:
     SEGM_DIR = "annot"
     IMAGE_EXT = '.jpg'
     MASK_EXT = '.png'
+    PATTERN = re.compile(r'(.+[.]\S+) (.+[.]\S+)?')
```
Comment (zhiltsov-max) [inline datumaro/plugins/camvid_format.py:64]: Please don't expect input datasets to be generated by Datumaro. Instead, consider meaningful relaxations to the format. Expect and fix little and insignificant errors you can think of.

I think, the pattern can be improved with `r(.+\.\S+)(?:\s+(.+\.\S+)?)\s*`
- to allow any number of whitespaces between entries
- to allow any number of trailing whitespaces (.strip() would also work)
- to allow lines like `a/b/c .d  \n`

File: datumaro/plugins/voc_format/extractor.py [hunk 5]
```diff
@@ -58,8 +58,24 @@ def _load_categories(dataset_path):
 
     @staticmethod
     def _load_subset_list(subset_path):
-        with open(subset_path) as f:
-            return [line.split()[0] for line in f]
+        subset_list = []
+        with open(subset_path, encoding='utf-8') as f:
+            for line in f:
+                dirname = osp.basename(osp.dirname(subset_path))
```
Comment (zhiltsov-max) [inline datumaro/plugins/voc_format/extractor.py:64]: I think this is better to be done with check of `self._task`. These files are not required to be inside the default task lists directory - it is just a default location.

File: datumaro/plugins/voc_format/extractor.py [hunk 6]
```diff
@@ -58,8 +58,24 @@ def _load_categories(dataset_path):
...
+                if dirname == VocPath.TASK_DIR[VocTask.person_layout]:
+                    objects = line.split('\"')
+                    if 1 < len(objects):
```
Comment (zhiltsov-max) [inline datumaro/plugins/voc_format/extractor.py:67]: `0 <` ? 1 will break this code.

**Review by yasakova-anastasia — COMMENTED** [review]
.

File: datumaro/plugins/camvid_format.py [hunk 7]
```diff
@@ -159,13 +160,17 @@ def _load_items(self, path):
         items = {}
         with open(path, encoding='utf-8') as f:
             for line in f:
-                objects = line.split()
+                search = CamvidPath.PATTERN.search(line.strip('\n'))
```
Comment (zhiltsov-max) [inline datumaro/plugins/camvid_format.py:163]: Why we strip only `\n`?

**Review by zhiltsov-max — DISMISSED** [review]

File: datumaro/plugins/camvid_format.py (hunk 7 above)
Comment (yasakova-anastasia) [reply datumaro/plugins/camvid_format.py:163]: We must always have a space after the first path. In [test without masks](https://github.com/openvinotoolkit/datumaro/blob/944e9f9f8ae2b10aea5eff9573e2de65844538c0/tests/test_camvid_format.py#L148), after `strip()`, we get the `test/a/b/1.png` without a space. As a result, search returns None.

Comment (yasakova-anastasia) [reply datumaro/plugins/camvid_format.py:163]: There is no such requirement. But I'm having difficulty with a regex that supports a string without a space. If we replace `+` with `*` in the regex (`r'(.+\.\S+)(?:\s*(.+\.\S+)?)\s*'`), we cannot find two paths.

Comment (zhiltsov-max) [reply datumaro/plugins/camvid_format.py:163]: Remove the last `\s*`

Comment (yasakova-anastasia) [reply datumaro/plugins/camvid_format.py]: This doesn't work as everything is recorded in the first group.

Comment (zhiltsov-max) [reply datumaro/plugins/camvid_format.py]: Did you try `(.+\.\S+)(?:\s+(.+\.\S+)?)?` ?

Comment (yasakova-anastasia) [reply datumaro/plugins/camvid_format.py]: Yes, also everything is recorded in one group

Comment (zhiltsov-max) [reply datumaro/plugins/camvid_format.py]: If we don't have a mask, it is what we need to obtain, isn't it?

Comment (yasakova-anastasia) [reply datumaro/plugins/camvid_format.py]: Yes. But everything is recorded in one group even we have a mask.

Comment (zhiltsov-max) [reply datumaro/plugins/camvid_format.py]: How about eager capture in the first group then? `(.+?\.\S+)(?:\s+(.+\.\S+)?)?`

Comment (yasakova-anastasia) [reply datumaro/plugins/camvid_format.py]: It works, thanks!

**Pull request merged**